commonutils/src/main/java/io/intrepid/commonutils/IntrepidAndroidUtils.java

```diff
@@ -0,0 +1,63 @@
+package io.intrepid.commonutils;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.support.annotation.NonNull;
+import android.util.DisplayMetrics;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+
+public class IntrepidAndroidUtils {
+    /**
+     * Hide the soft keyboard
+     *
+     * @param view The view that has the current focus
+     */
+    public static void hideKeyboard(View view) {
+        if (view == null) {
+            return;
+        }
+        InputMethodManager imm = (InputMethodManager) view.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+        imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
+    }
+
+    /**
+     * Show the soft keyboard
+     *
+     * @param view The view that has the current focus
+     */
+    public static void showKeyboard(View view) {
+        if (view == null) {
+            return;
+        }
+        InputMethodManager imm = (InputMethodManager) view.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+        imm.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT);
+    }
+
+    /**
+     * Converts dp unit to equivalent pixels, depending on device density.
+     *
+     * @param context Context to get resources and device specific display metrics
+     * @param dp      A value in dp (density independent pixels) unit. Which we need to convert into pixels
+     * @return A float value to represent px equivalent to dp depending on device density
+     */
+    public static float convertDpToPixel(@NonNull Context context, float dp) {
+        Resources resources = context.getResources();
+        DisplayMetrics metrics = resources.getDisplayMetrics();
+        return dp * (metrics.densityDpi / 160f);
```

Comment:
There are APIs in `TypedValue` to do this sort of thing.
`TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, metrics);`